Source/Events.Store.MongoDB/EventStore.cs

```diff
@@ -75,6 +79,8 @@ public async Task<CommittedEvents> CommitEvents(UncommittedEvents events, Cancel
                                 cancel).ConfigureAwait(false);
                             committedEvents.Add(committedEvent);
                             eventLogSequenceNumber++;
+
+                            _metrics.IncrementCommittedEvents(committedEvent);
                         }
 
                         return new CommittedEvents(committedEvents);
```

Comment:
And this is a transaction, so if one of the commit fails none of them goes through

Reply:
But it's in a for-loop and we do want to wait for the event to be committed fully until we increase the counter. If the `CommitEvent` method fails then we do not wan't to increment the counter, instead we catch the error later and increment the failed events counter.

Reply:
Yes, and that's why putting it in the end will work. If it throws an exception it will not increase the committed counter, and if one of the commits in the transaction fails it will also not increment the committed counter

Reply:
Oh whoops, hadn't refreshed the page so didn't see your comment about the transaction. Was confused as I thought that it was possible for some of the events to succeed and some of them to fail.